### PR TITLE
ref(stacktrace-link): Allow dismissing CTA

### PR DESCRIPTION
**Pull request by MeredithAnya**
**Context:**
We have a CTA for users to set up Stack Trace Linking. Right now it will always show up you have both GitHub or GitLab installed AND no code mapping set up for the project the issue belongs to.

We don't currently check roles but we should, members can't edit integration settings so they shouldn't see this CTA. 

**CTA Updates**
I've changed a couple of things about the CTA:
* Check for `org:integrations` access (everything above members)
* Made the CTA dismissible using the `/prompts-activity/` endpoint
* Added analytics for when users decide to dismiss
* Update the UI to accommodate the dismiss action

Screen Shot 2021-02-23 at 9 53 43 AM

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx [hunk 1]
```diff
@@ -74,6 +78,44 @@ class StacktraceLink extends AsyncComponent<Props, State> {
...
+    });
+  }
+
+  dismissPrompt() {
+    const {organization} = this.props;
+    promptsUpdate(this.api, {
+      organizationId: organization.id,
+      projectId: this.project?.id,
+      feature: 'stacktrace_link',
+      status: 'dismissed',
```
Comment (ceorourke) [inline src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx:105]: What's the difference between 'dismissed' and 'snoozed'? I assume the latter will pop back up after some period of time but the former is gone forever? If so is it gone forever for everyone or just that user?

Comment (MeredithAnya) [reply src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx:105]: @ceorourke yeah exactly, gone forever vs. gone for a certain amount of time. but its per the user, so I think just using dismiss in this case is probably fine.

**Review by ceorourke — APPROVED** [review]
lgtm but maybe a javascriptier person can review too

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx [hunk 2]
```diff
@@ -74,6 +78,44 @@ class StacktraceLink extends AsyncComponent<Props, State> {
...
+    const {organization} = this.props;
+
+    const prompt = await promptsCheck(this.api, {
+      organizationId: organization.id,
+      projectId: this.project?.id,
+      feature: 'stacktrace_link',
+    });
+
+    this.setState({
+      isDismissed: !prompt?.dismissedTime ? false : true,
```
Comment (scefali) [inline src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx:95]: @MeredithAnya I would recommend using the `promptIsDismissed` function which will make it easier to add snooze functionality later: https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/utils/promptIsDismissed.tsx#L5

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx [hunk 3]
```diff
@@ -97,6 +139,7 @@ class StacktraceLink extends AsyncComponent<Props, State> {
       showModal: false,
       sourceCodeInput: '',
       match: {integrations: []},
+      isDismissed: null,
```
Comment (scefali) [inline src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx:142]: @MeredithAnya Are we trying to differentiate between `null` and `false`? That could potentially lead to bugs if we aren't careful.

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx [hunk 4]
```diff
@@ -218,13 +270,19 @@ class StacktraceLink extends AsyncComponent<Props, State> {
...
+    const {isDismissed} = this.state;
+
     if (config && sourceUrl) {
       return this.renderMatchWithUrl(config, sourceUrl);
     }
     if (config) {
       return this.renderMatchNoUrl();
     }
 
+    if (isDismissed || isDismissed === null) {
```
Comment (scefali) [inline src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx:282]: @MeredithAnya would it be too much to add a new state variable called `promptIsLoaded` and make sure that is `true` before we check `isDismissed`?

File: src/sentry/utils/prompts.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
     "alert_stream": {"required_fields": ["organization_id"]},
     "sdk_updates": {"required_fields": ["organization_id"]},
     "suggest_mobile_project": {"required_fields": ["organization_id"]},
+    "stacktrace_link": {"required_fields": ["organization_id", "project_id"]},
```
Comment (scefali) [inline src/sentry/utils/prompts.py:7]: @MeredithAnya Does this mean a user who dismisses the prompt on one project will see it for another?

Comment (MeredithAnya) [reply src/sentry/utils/prompts.py:7]: That was my intention

Comment (scefali) [reply src/sentry/utils/prompts.py:7]: @MeredithAnya works for me unless we see users start complaining about it 😁

File: src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx (hunk 4 above)
Comment (MeredithAnya) [reply src/sentry/static/sentry/app/components/events/interfaces/stacktraceLink.tsx:282]: haha nah I can do that, I was really just looking to how this was done in other places, but I agree probs better to be explicit here

File: src/sentry/utils/prompts.py (hunk 5 above)
Comment (MeredithAnya) [reply src/sentry/utils/prompts.py:7]: @scefali yeah we can always adjust, especially when it goes to GA, but I think it makes more sense on a project level for now

**Pull request merged**